app/manager-frontend/src/pages/ActivitySource/index.jsx

```diff
@@ -113,21 +125,49 @@ export const ActivitySource = ({ activitySource, action, id }) => {
                   name={"ResourceId"}
                   type="ResourceId"
                 />
-                <Button
-                  w="full"
-                  leftIcon={<FaArrowRight />}
-                  colorScheme="blue"
-                  type="submit"
-                  disabled={isSubmitting}
-                  isLoading={isSubmitting}
-                >
-                  {submitText}
-                </Button>
+                <Flex>
+                  <Button
+                    w="full"
+                    leftIcon={<FaArrowRight />}
+                    colorScheme="blue"
+                    type="submit"
+                    disabled={isSubmitting}
+                    isLoading={isSubmitting}
+                    mx={id ? 4 : 0}
+                  >
+                    {submitText}
+                  </Button>
+                  {id && (
+                    <Button w="full" colorScheme="red" onClick={onOpen} mx={4}>
+                      Delete
+                    </Button>
+                  )}
+                </Flex>
               </VStack>
             </Form>
           )}
         </Formik>
       </VStack>
+      <Modal isOpen={isOpen} onClose={onClose}>
```

Comment:
♻️ Factor this modal out into it's own component, instead of having it twice identically in this page and the homepage.